pkg/sentry/pgalloc/pgalloc.go

```diff
@@ -579,7 +579,12 @@ func (f *MemoryFile) AllocateAndFill(length uint64, kind usage.MemoryKind, r saf
 	if err != nil {
 		return memmap.FileRange{}, err
 	}
-	dsts, err := f.MapInternal(fr, hostarch.Write)
+	access := hostarch.Write
+	if kind == usage.Tmpfs {
+		// We assume that NoAccess means to populate memory
+		access = hostarch.NoAccess
+	}
```

Comment:
ehh... this feels hacky... This only works because MemoryFile.MapInternal() only cares about executability and we are only passing hostarch.Write.

I think there are 2 things we could do here:
1. Add a new method on MemorFile called `func (f *MemoryFile) mapInternal(fr memmap.FileRange, at hostarch.AccessType, populate bool) (safemem.BlockSeq, error) {}` and make the **exported** `MapInternal()` simply call into this. In AllocateAndFill(), you can call `dsts, err := f.mapInternal(fr, access, kind == usage.Tmpfs)`
2. Update the `memmap.File.MapInternal` interface method to take in a simple struct `memmap.MapInternalOptions`. This struct can contain 2 fields for now: `at hostarch.AccessType` and `populate bool`. This way, `MapInternal()` API function is more extensible. This will need a larger refactor of all usages.

I will leave it up to you.

Reply:
thx, I think the first one will be better, and it's done